### ISSUE TITLE
Improve performance

### ISSUE THREAD

**Issue by rikatz** [rikatz/kubepug#62]
Since the new library refactor it seems pug is pretty slower, as the following:

```
# before refactoring
real    0m2,050s
user    0m0,453s
sys     0m0,091s
```

```
# After refactoring
real    0m30,541s
user    0m0,866s
sys     0m0,126s
```

This needs to be investigated before cutting a new release

**Comment by dguyhasnoname** [comment]
Other way around find deprecations: https://github.com/dguyhasnoname/k8s-day2-ops/blob/master/upgrade/k8s-deprecations.sh

**Comment by rikatz** [comment]
So it seems that always passing the pure config and always creating a new client, instead of creating a bigger client connection that could be used widely seems much faster.

Specifically [this change](https://github.com/rikatz/kubepug/pull/48/files#diff-6219b4699acdbd3cf8c164bf179cb62cL35-L40)  is what's causing Kubepug to take 40s instead of 2s to populate the Map.

Will continue to debug

**Comment by rikatz** [comment]
Fixed in #66

**Comment by rikatz** [comment]
Performance improvements:
* Create a query into kubernetes every time instead of passing this as a pointer
* Fix a bug where the comparison between objects in the walker are always false
